src/loader.ts

```diff
@@ -364,11 +412,11 @@ interface ModuleDefinitionArguments extends Array<any> {
 		array && array.forEach(callback);
 	}
 
-	function mix(target: {}, source: {}): {} {
+	function mix<T extends {}>(target: {}, source: {}): T {
```

Comment:
Agreed, we should assign and offload to native assign actually, depending on how easy it is.  We should open an issue.